gitlab/resource_gitlab_project.go

```diff
@@ -11,6 +13,8 @@ import (
 	gitlab "github.com/xanzy/go-gitlab"
 )
 
+var errProjectPushRulesUnsupported = errors.New("Project push rules are not supported in your version of GitLab")
```

Comment:
```suggestion
var errProjectPushRulesUnsupported = errors.New("project push rules are not supported in your version of GitLab")
```

Reply:
In other providers I have seen them use capitalized errors for errors that are returned from the CRUD functions, because they are user-facing. I moved this error back to inline to ease the confusion. Let me know if you still want me to change it.